### PR TITLE
✨ Added a log collector implementation for Metal3

### PR DESCRIPTION
**Pull request by huutomerkki**
Now only collects qemu serial logs.

**What this PR does / why we need it**:

This PR contains the basic implementation of [CAPI log collector,](https://github.com/kubernetes-sigs/cluster-api/blob/ddb18986a0e325c8a877b19be817079938e32bae/test/framework/cluster_proxy.go#L105) and the collecting of serial logs. In the future, different logs can also be collected.

The collected logs can be checked in the tarball from the path: `e2e_artifacts/_artifacts/machines/`

### CODE REVIEW

**Comment by mboukhalfa** [comment]
Can you please squash commits

File: test/e2e/common.go [hunk 1]
```diff
@@ -554,3 +556,65 @@ func Metal3MachineToBmhName(m3machine infrav1.Metal3Machine) string {
...
+
+	for _, log := range serialLogs {
+		cmd := exec.Command("sudo", "cp", log, qemuFolder) // #nosec G204:gosec
+		if output, err := cmd.Output(); err != nil {
+			return fmt.Errorf("something went wrong when executing '%s': %w, output: %s", cmd.String(), err, output)
+		}
+		cmd = exec.Command("sudo", "chmod", "750", path.Join(qemuFolder, filepath.Base(log))) // #nosec G204:gosec
+		if output, err := cmd.Output(); err != nil {
+			return fmt.Errorf("error changing file permissions after copying: %w, output: %s", err, output)
+		}
```
Comment (mboukhalfa) [inline test/e2e/common.go:608]: Why we bother change chmod after copying ?

File: test/e2e/integration_test.go [hunk 2]
```diff
@@ -1,79 +1,33 @@
...
-		}, intervals...)
-		numberOfAvailableBMHs := numberOfWorkers + numberOfControlplane
-		intervals = e2eConfig.GetIntervals(specName, "wait-bmh-deprovisioning-available")
-		WaitForNumBmhInState(ctx, bmov1alpha1.StateAvailable, WaitForNumInput{
-			Client:    bootstrapClient,
-			Options:   []client.ListOption{client.InNamespace(namespace)},
-			Replicas:  numberOfAvailableBMHs,
-			Intervals: intervals,
-		})
+		targetCluster, _ = createTargetCluster(k8sVersion)
```
Comment (mboukhalfa) [inline test/e2e/integration_test.go:27]:
```suggestion
		createTargetCluster(k8sVersion)
		By("PASSED!")
```
I guess targetCluster is not used anywhere

File: test/e2e/integration_test.go [hunk 3]
```diff
@@ -1,79 +1,33 @@
...
+	BeforeEach(func() {
+		osType := strings.ToLower(os.Getenv("OS"))
+		Expect(osType).ToNot(Equal(""))
+		validateGlobals(specName)
+
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+
 	It("CI Test Provision", func() {
```
Comment (mboukhalfa) [inline test/e2e/integration_test.go:22]:
```suggestion
	It("Should create a workload cluster", func() {
	By("Creating a workload cluster")
```

**Review by mboukhalfa — COMMENTED** [review]
I added some labeling to follow capi Style and I guess separating the collector changes and the basic test in two different commit I will make it easier

File: test/e2e/integration_test.go (hunk 2 above)
Comment (huutomerkki) [reply test/e2e/integration_test.go:27]: Not completely sure, but isn't it used in the AfterEach block as a parameter?

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**